src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests/ServiceControllerTests.cs

```diff
@@ -218,43 +220,77 @@ public void GetDevices()
 
             for (int i = 0; i < controller.DependentServices.Length; i++)
             {
-                var dependent = controller.DependentServices[i];
-                Assert.True(dependent.ServiceName.StartsWith(_testService.DependentTestServiceNamePrefix));
-                Assert.True(dependent.DisplayName.StartsWith(_testService.DependentTestServiceDisplayNamePrefix));
+                var dependent = AssertHasDependent(controller, _testService.DependentTestServiceNamePrefix + i, _testService.DependentTestServiceDisplayNamePrefix + i);
                 Assert.Equal(ServiceType.Win32OwnProcess, dependent.ServiceType);
-                Assert.Equal(0, dependent.DependentServices.Length);
+
+                // Assert that this dependent service is depended on by all the test services created after it
+                Assert.Equal(ExpectedDependentServiceCount - i - 1, dependent.DependentServices.Length);
+
+                for (int j = i + 1; j < ExpectedDependentServiceCount; j++)
+                {
+                    AssertHasDependent(dependent, _testService.DependentTestServiceNamePrefix + j, _testService.DependentTestServiceDisplayNamePrefix + j);
+                }
+
+                // Assert that the dependent service depends on the main test service
+                AssertDependsOn(dependent, _testService.TestServiceName, _testService.TestServiceDisplayName);
+
+                // Assert that this dependent service depends on all the test services created before it
+                Assert.Equal(i + 1, dependent.ServicesDependedOn.Length);
+
+                for (int j = i - 1; j >= 0; j--)
+                {
+                    AssertDependsOn(dependent, _testService.DependentTestServiceNamePrefix + j, _testService.DependentTestServiceDisplayNamePrefix + j);
+                }
             }
         }
 
-        [Fact]
-        public void ServicesDependedOn()
+        public void Dispose()
         {
-            // The test service creates a number of dependent services, each of these should depend on the test service
-            var controller = new ServiceController(_testService.TestServiceName);
-            Assert.True(controller.DependentServices.Length > 0);
+            _testService.DeleteTestServices();
+        }
 
-            for (int i = 0; i < controller.DependentServices.Length; i++)
+        private static ServiceController AssertHasDependent(ServiceController controller, string serviceName, string displayName)
+        {
+            var dependent = FindService(controller.DependentServices, serviceName, displayName);
+            if (dependent == null)
             {
-                var dependent = controller.DependentServices[i];
-                Assert.True(dependent.ServicesDependedOn.Length == 1);
+                Assert.True(false, string.Format("Expected service {0} to have dependent service {1}", controller.ServiceName, serviceName));
```

Comment:
I see us doing things like this in our tests a fair bit, e.g. instead of using Assert.NotNull, checking for null and then using Assert.True(false, ...), in order to include an error message.  It complicates our tests, though I understand in some cases the value of having the message (including here).  We might consider adding some helpers like AssertNotNull(obj, message) that we put in some shared library, or better yet, work with xunit to get the appropriate overloads added.